lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -158,21 +158,39 @@ protected static function translate( $theme_json, $domain = 'default' ) {
 	 * @return WP_Theme_JSON_Gutenberg Entity that holds core data.
 	 */
 	public static function get_core_data() {
+		$can_use_cached = ! wp_is_development_mode( 'theme' );
+		if ( $can_use_cached ) {
+			$cache_key = 'core_data';
+			if ( null === static::$core ) {
+				$cache_value = static::get_cache_data( $cache_key );
+				if ( $cache_value ) {
+					// Defined in this function below.
+					$theme_json   = apply_filters( 'wp_theme_json_data_default', $cache_value );
```

Comment:
I can try caching `get_data`, but I feel it will negate all our effort as most of the performance regression came from `WP_Theme_JSON_Data_Gutenberg` constructor call

Reply:
> most of the performance regression came from WP_Theme_JSON_Data_Gutenberg constructor call
I thought that might be the case.

Let's handle backporting https://github.com/WordPress/wordpress-develop/commit/6ae489ccaeb1e426d496183c2c1c997295f05d42 separately. I'll get a PR open.